packages/webdriverio/src/utils.js

```diff
@@ -414,3 +415,33 @@ export const enhanceElementsArray = (elements, parent, selector, foundWith = '$$
  * @param {string} automationProtocol
  */
 export const isStub = (automationProtocol) => automationProtocol === './protocol-stub'
+
+export const getAutomationProtocol = async (config) => {
+    /**
+     * don't modify automation protocol if hostname is set as it is very likely
+     * that "webdriver" will be used
+     */
+    if (config.hostname || (config.user && config.key)) {
+        return config.automationProtocol
+    }
+
+    /**
+     * make a head request to check if a driver is available
+     */
+    const driverEndpointHeaders = await new Promise((resolve) => {
+        const req = http.request({
+            method: 'GET',
+            host: 'localhost',
+            port: config.port || 4444,
+            path: '/status'
```

Comment:
isn't config.path ignored in such case?

Reply:
I will remove `config.port` - if nothing is specified we need to assume that the driver is on `localhost:4444`.

Reply:
Yeah, but path /status is not necessarily valid. Can it be /wd/hub/path ?

Reply:
`/status` is defined by the WebDriver spec and therefor has to be there